packages/hoppscotch-app/components/collections/graphql/AddRequest.vue

```diff
@@ -0,0 +1,78 @@
+<template>
+  <SmartModal
+    v-if="show"
+    dialog
+    :title="$t('request.new')"
+    @close="$emit('hide-modal')"
+  >
+    <template #body>
+      <div class="flex flex-col px-2">
+        <input
+          id="selectLabelGqlAddRequest"
+          v-model="name"
+          v-focus
+          class="input floating-input"
+          placeholder=" "
+          type="text"
+          autocomplete="off"
+          @keyup.enter="addRequest"
+        />
+        <label for="selectLabelGqlAddRequest">
+          {{ $t("action.label") }}
+        </label>
+      </div>
+    </template>
+    <template #footer>
+      <span>
+        <ButtonPrimary :label="$t('action.save')" @click.native="addRequest" />
+        <ButtonSecondary
+          :label="$t('action.cancel')"
+          @click.native="hideModal"
+        />
+      </span>
+    </template>
+  </SmartModal>
+</template>
+
+<script setup lang="ts">
+import { ref, watch } from "@nuxtjs/composition-api"
+import { useI18n, useToast } from "~/helpers/utils/composables"
+import { useGQLRequestName } from "~/newstore/GQLSession"
+
+const toast = useToast()
+const t = useI18n()
+
+const props = defineProps<{
+  show: boolean
+  folderPath?: string
+}>()
+
+const emit = defineEmits<{
+  (e: "hide-modal"): void
+  (e: "add-request", v: object): void
+}>()
+
+const name = ref("")
+
+watch(props, (v) => {
+  if (v.show) {
+    name.value = useGQLRequestName().value
```

Comment:
Do not use composables like this, because many composables are only intended to be called during setup and not afterwards in the lifecycle.
Also, for this function, it creates a sort of a leak, as each time, a stream connection is created again when the watcher is called.

You don't have to subscribe to a stream of values here (which is what the composable does), we only want the value at that instant.
Hence, instead of doing this, we can just get the present GQL Session from `getGQLSession()` and then get the name from its request.
That gives us the value at that instant, which is what we need.

Reply:
> Also, for this function, it creates a sort of a leak, as each time, a stream connection is created again when the watcher is called.

Ahh yes, I didn't look up the function.